dotnet/src/Connectors/Connectors.AI.OpenAI/KernelConfigOpenAIExtensions.cs

```diff
@@ -393,11 +395,27 @@ public static class KernelConfigOpenAIExtensions
 
     #endregion
 
-    private static HttpClient CreateHttpClient(this IDelegatingHandlerFactory handlerFactory,
-        ILogger? logger)
+    /// <summary>
+    /// Retrieves an instance of HttpClient.
+    /// </summary>
+    /// <param name="kernel">The kernel.</param>
+    /// <param name="httpClient">An optional pre-existing instance of HttpClient.</param>
+    /// <param name="logger">An optional logger.</param>
+    /// <returns>An instance of HttpClient.</returns>
+    private static HttpClient GetHttpClient(IKernel kernel, HttpClient? httpClient, ILogger? logger)
     {
-        var retryHandler = handlerFactory.Create(logger);
-        retryHandler.InnerHandler = new HttpClientHandler { CheckCertificateRevocationList = true };
-        return new HttpClient(retryHandler);
+        if (httpClient == null)
+        {
+            var retryHandler = kernel.Config.HttpHandlerFactory.Create(logger ?? kernel.Log);
```

Comment:
Pre-existing your PR, but it seems like we're creating our own variant of IHttpClientFactory here?  What's the reason for that?

Reply:
Yep, looks like it. Later, we may consider extracting it into a separate factory class that can be used by all SK SDK components that need to use HTTP client.

Reply:
@SergeyMenshykh @stephentoub Would [SocketsHttpHandler ](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler?view=net-7.0) be appropriate to use here (as an assembly-level fallback, if httpClient args remain optional and caller doesn't provide one?
As an alternative to HttpClientFactory, without the DI stuff, it still provides common settings and protection against port exhaustion.

Reply:
Yes... but not while targeting netstandard2.0.  If/when the assembly multitargets, it could use a singleton HttpClientHandler for netstandard2.0 and a singleton SocketsHttpHandler for net6.0+.